tests/functional/watcherapi_controller_test.go

```diff
@@ -0,0 +1,134 @@
+package functional
+
+import (
+	. "github.com/onsi/ginkgo/v2" //revive:disable:dot-imports
+	. "github.com/onsi/gomega"    //revive:disable:dot-imports
+
+	//revive:disable-next-line:dot-imports
+	condition "github.com/openstack-k8s-operators/lib-common/modules/common/condition"
+	. "github.com/openstack-k8s-operators/lib-common/modules/common/test/helpers"
+	mariadbv1 "github.com/openstack-k8s-operators/mariadb-operator/api/v1beta1"
+	watcherv1beta1 "github.com/openstack-k8s-operators/watcher-operator/api/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+)
+
+var _ = Describe("WatcherAPI controller with minimal spec values", func() {
+	When("A Watcher instance is created from minimal spec", func() {
+		BeforeEach(func() {
+			DeferCleanup(th.DeleteInstance, CreateWatcherAPI(watcherTest.Instance, MinimalWatcherSpec))
+		})
+
+		It("should have the Spec fields defaulted", func() {
+			WatcherAPI := GetWatcherAPI(watcherTest.Instance)
+			Expect(WatcherAPI.Spec.DatabaseInstance).Should(Equal("openstack"))
+			Expect(WatcherAPI.Spec.DatabaseAccount).Should(Equal("watcher"))
+			Expect(WatcherAPI.Spec.Secret).Should(Equal("osp-secret"))
+			Expect(WatcherAPI.Spec.PasswordSelectors).Should(Equal(watcherv1beta1.PasswordSelector{Service: "WatcherPassword"}))
+		})
+
+		It("should have the Status fields initialized", func() {
+			WatcherAPI := GetWatcherAPI(watcherTest.Instance)
+			Expect(WatcherAPI.Status.ObservedGeneration).To(Equal(int64(0)))
+		})
+
+		It("should have a finalizer", func() {
+			// the reconciler loop adds the finalizer so we have to wait for
+			// it to run
+			Eventually(func() []string {
+				return GetWatcherAPI(watcherTest.Instance).Finalizers
+			}, timeout, interval).Should(ContainElement("openstack.org/watcherapi"))
+		})
+
+	})
+})
+
+var _ = Describe("WatcherAPI controller", func() {
+	When("A WatcherAPI instance is created", func() {
+		BeforeEach(func() {
+			DeferCleanup(th.DeleteInstance, CreateWatcherAPI(watcherTest.Instance, GetDefaultWatcherAPISpec()))
+		})
+
+		It("should have the Spec fields defaulted", func() {
+			WatcherAPI := GetWatcherAPI(watcherTest.Instance)
+			Expect(WatcherAPI.Spec.DatabaseInstance).Should(Equal("openstack"))
+			Expect(WatcherAPI.Spec.DatabaseAccount).Should(Equal("watcher"))
+			Expect(WatcherAPI.Spec.Secret).Should(Equal("test-osp-secret"))
+		})
+
+		It("should have the Status fields initialized", func() {
+			WatcherAPI := GetWatcherAPI(watcherTest.Instance)
+			Expect(WatcherAPI.Status.ObservedGeneration).To(Equal(int64(0)))
+		})
+
+		It("should have unknown Conditions initialized", func() {
+			th.ExpectCondition(
+				watcherTest.Instance,
+				ConditionGetterFunc(WatcherAPIConditionGetter),
+				condition.ReadyCondition,
+				corev1.ConditionFalse,
```

Comment:
false or unknown ?

Reply:
that should be false indeed, I fixed the description